frontend/angular/style-guide.md

```diff
@@ -0,0 +1,972 @@
+
+## Properties order
+
+- decorators
+  - inputs
+  - outputs
+  - others
+- public properties
+- public readonly properties
+- private properties
+- private readonly properties
+- setter and getter properties (accessors)
+  - public
+  - private
+
+```ts
+@Component({
+  selector: 'app-cool',
+  template: `<div>I'm a nice App</div>`,
+})
+export class AppComponent {
+  @Input() foo;
+  @Ouput() fooChange = new EventEmiter();
+  isCool = true;
+  private name = 'wolox';
+  readonly place = 'world';
+
+  get woo() {
+    return this.anyValue;
+  }
+
+  set woo(value) {
+    this.anyValue = value;
+  }
+
+  constructor() {}
+}
+```
+
+**Avoid**
+
+```ts
+@Component({
+  selector: 'app-cool',
+  template: `<div>I'm a nice App</div>`,
+})
+export class AppComponent {
+  get woo() {
+    return this.anyValue;
+  }
+
+  set woo(value) {
+    this.anyValue = value;
+  }
+
+  private name = 'wolox';
+  isCool = true;
+  readonly place = 'world';
+  @Ouput() fooChange = new EventEmiter();
+  @Input() foo;
+
+  constructor() {}
+}
+```
+
+
+## Components communication and interaction
+
+   1. Two-Way Data Binding (Banana In a Box [🍌]).
+   2. Decorators instead of properties of the Metadata.
+   3. Operations in the Template.
+   
+
+#### 1) Two-Way Data Binding (Banana In a Box [🍌]).
+
+
+To achieve this, you must add `Change` suffix to `@Output()` name, and the same name as the `@Input()`.
+
+```html
+<my-comp [(prop)]="val"> </my-comp>
+```
+
+This is the syntax for Two-way binding, namely, the property binding and event binding:
+
+```html
+<my-comp [prop]="val" (propChange)="val=$event"> </my-comp>
+```
+
+**Avoid**
+
+```html
+<my-comp [prop]="val" (prop)="val=$event"> </my-comp>
+```
+
+#### 2) Decorators instead of properties of the Metadata.
+
+The Component and Directive decorators have some particularity, they receive configurations in its metadata you could write ans manage more easily in the class.
+
+```ts
+@[Component|Directive]({
+  inputs: [[enums]],
+  outputs: [[enums]],
+  hosts: [[enums]]
+})
+```
+
+This configuration is poorly readable. It's preferable to use the corresponding decorators.
+
+- `@Inputs()`
+- `@Outputs()`
+- `@HostBinding()`
+- `@HostListener()`
+
+#### 3) Operations in the Template.
+
+The Operators in the template are often confusing. Use getters and your code will be cleaner and more declarative.
+
+```ts
+@Component({
+  selector: 'app-cool',
+  template: `<div>The result is {{ nicerResult }}</div>`,
+})
+export class AppComponent {
+  @Input() foo;
+
+  get nicerResult() {
+    return (this.foo * 32) / 10 + 300;
+  }
+
+  constructor() {}
+}
+```
+
+**Avoid**
+
+```ts
+@Component({
+  selector: 'app-cool',
+  template: `<div>The result is {{ (foo * 32) / 10 + 300 }}</div>`,
+})
+export class AppComponent {
+  @Input() foo;
+
+  constructor() {}
+}
+```
+
+## Aliasing inputs and outputs
+
+**Avoid** input and output aliases except when it serves an important purpose.
```

Comment:
Can we add an example of an `important purpose `?

Reply:
I going to remove this part